02/ps-optim-example.Rmd

```diff
@@ -0,0 +1,184 @@
+- value: the value of the objective function when passed these optimal parameters
+- counts: the number of times the objective function and gradient needed to be calculated for a new set of possible parameters. (in this case, optim didn't use any gradients to optimize)
+- convergence: an integer code either saying that the optimization was successful (zero) or that an error was encountered (nonzero values)
+- message: any additional messages relevant to the optimization
+
+`lm`:
+```{r}
+class(fit)
+```
+
+lm returns an "lm" object that encodes the arguments passed to lm and the optimal coefficients that it found.
+
+* How different were the parameter estimates? Can compare in a nice table (use`kable`)?
+```{r}
+
+# Removing the names from the coefficient lists for fit
+coeffs <- fit$coefficients
+names(coeffs) <- NULL
+
+# Making the dataframe to be converted into a table
+method_names <- c("optim, par = (0, 1)",
+                  "optim, par = (10, 10)",
+                  "optim, par = (0, -10)",
+                  "lm")
+par1 <- c(unlist(optim_output, use.names = FALSE)[1], 
+          unlist(alt_output, use.names = FALSE)[1],
+          unlist(alt_output_2, use.names = FALSE)[1],
+          coeffs[1])
+par2 <- c(unlist(optim_output, use.names = FALSE)[2], 
+          unlist(alt_output, use.names = FALSE)[2],
+          unlist(alt_output_2, use.names = FALSE)[2],
+          coeffs[2])
+
+all_pars <- data.frame(method_names, par1, par2)
+
+# Producing the table with kable
+all_pars %>%
+  kbl() %>%
+  kable_styling()
+
+```
+The estimated parameter values were all quite similar, although there is considerably more variability (at least in order of magnitude) for the first parameter (the intercept).
+
+* How do the predicted results using the different par estimates compare with one another?
+
+Let's add a couple columns testing different predictions for different x-values from these different estimates:
+
+```{r}
+predicted_data <- all_pars
+predicted_data$x_neg100 <- all_pars$par1 + all_pars$par2 * (-100)
+predicted_data$x_neg50 <- all_pars$par1 + all_pars$par2 * (-50)
+predicted_data$x_0 <- all_pars$par1
+predicted_data$x_50 <- all_pars$par1 + all_pars$par2 * (50)
+predicted_data$x_100 <- all_pars$par1 + all_pars$par2 * (100)
+
+# Producing the table with kable
+predicted_data %>%
+  kbl() %>%
+  kable_styling()
+
+```
+
+These estimates are all quite similar, although the par = (10, 10) pars predict values that are noticeably off from the other parameter combinations.
+
+* Evaluate parameter fit performance (compare predicted values of y with the original values using MSE)
+
+```{r}
+
+# Calculating MSE
+all_pars$MSE <- objective_function(data, c(all_pars$par1, all_pars$par2))
+
+# Producing the table with kable
+all_pars %>%
+  kbl() %>%
+  kable_styling()
+
+```
+
+Despite the slightly different parameter values, all of the parameter combinations produce the same MSE, suggesting that they all exhibit very similar performance.
```

Comment:
Very nicely done! Great tables, could you add a graph visually comparing the original data & then the two different fits all in one plot? 
The training data should be points and the two fits should be added to the plot as lines